### DISCUSSION TITLE
Problem with Presence in 24G tuya radar

### DISCUSSION THREAD

**Discussion by beliy82** [Koenkk/zigbee2mqtt#23484] [Devices]
Hi, just added Tuya RT_ZCZ03Z Presence sensor,
and the illuminance work great, target distance is working, but the presence stays always on false... any idea?

**Comment by beliy82** [comment]
same here.... hoped the new update will fix it, but no

**Reply by oritor91** [reply]
I also have the same issue. Help needed if anyone figured it out.

**Comment by beliy82** [comment]
any news?

**Comment by S12Link** [comment]
Same here. All parameters work correctly except presence, always false.
Is it a zigbee2MQTT bug?

**Reply by S12Link** [reply]
Since the target distance works, I have created an automation based on it,  turning the lights on and off depending on this target distance.

I have carried out several tests to find out the maximum distance at which it does not detect and, based on this, I turn the light on and off, adding a delay to the disconnection to avoid false negatives.

It's a trap or a trick XD but at the moment it works quite well and I can use the device for whatever I bought it for.
I hope they can fix this bug, whether it's Tuya or Zigbee2MQTT.

Has anyone tried it with a Tuya zigbee hub? Device failure could be ruled out and Zigbee2MQTT developers informed.

**Reply by beliy82** [reply]
i am with homekit and homebridge, i cannot make automation based on distance, but with zemismart it works fine, in smartlife app

**Reply by S12Link** [reply]
Then it must be a zigbee2Mqtt problem. Do the developers read this thread? It would be interesting if they reviewed it because perhaps the solution could come with an update.

**Reply by 6PATyCb** [reply]
While we wait for the fix, you can try this converter
[RT_ZCZ03Z.zip](https://github.com/user-attachments/files/17450541/RT_ZCZ03Z.zip)

**Reply by beliy82** [reply]
how to use it?

**Reply by S12Link** [reply]
Hello, thank you very much, I didn't know that this type of external files existed.

I have been reading about the issue in several Github threads, seeing that there are more people with this problem on similar devices.
The converter you attached makes presence detection work, but sometimes it goes crazy and gives false positives.
The two states that appear to me are: 0.- Unknown, 1.- Present.

![Captura desde 2024-10-20 19-17-03](https://github.com/user-attachments/assets/1d441283-db13-41a3-aea5-5ca431380955)
![Captura desde 2024-10-20 19-20-01](https://github.com/user-attachments/assets/524196dd-582d-490d-aa09-7f6b06154fb1)

The presence option does not appear in the automations in home assistant either, only when you configure it as an entity, and the options are unknown, any state and not available.
I use it with the value "Not available" to do the automation.

![Captura desde 2024-10-20 19-26-29](https://github.com/user-attachments/assets/cf98dfa5-5859-4501-8f82-ad3218ca544e)

To prevent the light from turning on when giving a false positive, I have added the target distance as a condition.
We are still testing, let's see when they solve the problem, but thank you very much for the contribution friend.

To use these external files (converters), I pass the link:
https://github.com/Koenkk/zigbee2mqtt/issues/17483#issuecomment-1527902715

**Comment by S12Link** [comment]
Hello, thank you very much, I didn't know that this type of external files existed.

I have been reading about the issue in several Github threads, seeing that there are more people with this problem on similar devices.
The converter you attached makes presence detection work, but sometimes it goes crazy and gives false positives.
The two states that appear to me are: 0.- Unknown, 1.- Present.

![Captura desde 2024-10-20 19-17-03](https://github.com/user-attachments/assets/1d441283-db13-41a3-aea5-5ca431380955)
![Captura desde 2024-10-20 19-20-01](https://github.com/user-attachments/assets/524196dd-582d-490d-aa09-7f6b06154fb1)

The presence option does not appear in the automations in home assistant either, only when you configure it as an entity, and the options are unknown, any state and not available.
I use it with the value "Not available" to do the automation.

![Captura desde 2024-10-20 19-26-29](https://github.com/user-attachments/assets/cf98dfa5-5859-4501-8f82-ad3218ca544e)

To prevent the light from turning on when giving a false positive, I have added the target distance as a condition.
We are still testing, let's see when they solve the problem, but thank you very much for the contribution friend.

To use these external files (converters), I pass the link:
https://github.com/Koenkk/zigbee2mqtt/issues/17483#issuecomment-1527902715

**Comment by 6PATyCb** [comment]
This is not my script. I found it in one of the forums and fixed it so that it works on the latest version of z2m. The script is not perfect, some things in it still do not work. It also seems to me that you and I can have different devices that have a common chip and are recognized the same way, but they can have different firmware and different behavioral bugs. My device looks like this:
![изображение](https://github.com/user-attachments/assets/b947f31c-864f-41a6-89a2-af08c4f7db5b)

**Reply by S12Link** [reply]
In appearance, it is the same. It also has green wall plugs 😂. 
I think the false positives were due to a curtain moving with the air, so that's not a problem. 
The rest works, except the indicator, which gives an error in the zigbee2Mqtt log. 
But it works much better than before, and the errors it has are negligible. 
Thank you very much, now at least it works as it should.

**Comment by 6PATyCb** [comment]
[RT_ZCZ03Z_v2.zip](https://github.com/user-attachments/files/17451993/RT_ZCZ03Z_v2.zip)

I corrected the script so that the LED would work and so that the presence would be true/false. For the script to work properly, you need to forcefully remove the device and add it again (after connecting the script file):
![изображение](https://github.com/user-attachments/assets/72cab0c6-c539-4418-a61b-b6e829872160)

**Comment by 6PATyCb** [comment]
And now version 3, which looks better in HA.
[RT_ZCZ03Z_v3.zip](https://github.com/user-attachments/files/17452023/RT_ZCZ03Z_v3.zip)

**Comment by 6PATyCb** [comment]
And version 4, with worked reset button
[RT_ZCZ03Z_v4.zip](https://github.com/user-attachments/files/17452796/RT_ZCZ03Z_v4.zip)

**Reply by S12Link** [reply]
Thanks man, you're great.
I have used the V4 script and it works perfectly, just as you said, both the presence and the indicator.
Thank you very much man.

**Comment by oritor91** [comment]
I also use homebridge with zigbee2mqtt plugin (running on raspberry pi 4) -
how can I use the RT_ZCZ03Z_v4 script? would love for a short explanation.

**Reply by 6PATyCb** [reply]
I think it must work for you. 
You need:
1. upload script file to z2m data folder:
![изображение](https://github.com/user-attachments/assets/4ce65b30-323c-4964-990c-452ae915d447)

2. include file to z2m (you can do this over web):
![изображение](https://github.com/user-attachments/assets/38e83393-aa3b-46dd-a9bc-354546f2459f)
3. restart z2m(If all good take a look at z2m start log):
![изображение](https://github.com/user-attachments/assets/8139871a-c551-4d7d-ae31-c6241475d3e0)
4. profit! Now your device must work fine:
![изображение](https://github.com/user-attachments/assets/702a0d6a-fff6-4f56-b9b8-a903d20a595e)
device sensor real time changes will be visible at 'State' tab:
![изображение](https://github.com/user-attachments/assets/d9b4a30f-2f05-4720-a413-a5281da4ecc9)

**Comment by oritor91** [comment]
It worked! Thank you so much

**Comment by S12Link** [comment]
Hello again. After working fine all this time, now I'm having problems again.
The detector remains "present" all the time, as if it were blocked.
I tried resetting it, without success.
I tried leaving it without voltage for a while and then resetting it, without success.
I tried removing it from zigbee2mqtt and rebinding it. This seemed to work, but it only lasted two days, then the same thing happened again, always "presence".
Does it happen to anyone else?
Are there new versions of the external converter?

On the zigbee2MQTT page (devices) there is no problem with this detector and no link to github with the discussion, but here we are, with operating problems.

**Reply by 6PATyCb** [reply]
What version of z2m are you using? I haven't used this device for a long time because I bought another one. I didn't make a new version of the converter.

**Reply by S12Link** [reply]
Hello, sorry for the delay in responding.
I'll give you an image with the z2m data.

![Captura desde 2024-11-28 18-20-19](https://github.com/user-attachments/assets/a802cbef-737d-4427-89f7-99c112c83fc1)

**Reply by 6PATyCb** [reply]
I updated to the same version and got the old device. It works fine with converter. Maybe you have something wrong with the converter. Pay attention to my comments above. In the log when starting z2m there should be information about the successful launch of the converter. If I remember correctly, when the converter does not work, all distances are displayed in mm, and when it works in m.

**Reply by 6PATyCb** [reply]
Don't forget set "None delay time" to "0" before presence test
![изображение](https://github.com/user-attachments/assets/f875dd74-4d1d-4b53-8b57-0a8fd8d450e1)

**Reply by S12Link** [reply]
OK thanks. I Change the device location and restart it. Now it works correctly. I'm watching to see if it fails again, but it's possible that in the old location it malfunctioned due to electrical interference with the detector's data cable. Thank you very much for your effort.
